mlflow/sklearn/__init__.py

```diff
@@ -948,6 +1038,53 @@ def patched_fit(original, self, *args, **kwargs):
                     FLAVOR_NAME, class_def, func_name, patched_fit, manage_run=True,
                 )
 
+        for func_name in ["predict", "transform", "predict_proba"]:
+            if hasattr(class_def, func_name):
+                safe_patch(
+                    FLAVOR_NAME, class_def, func_name, patched_predict, manage_run=False,
+                )
+
+    from sklearn import metrics
+    # Note:
+    #  don't need patch model.score() because internally model.score() will call the concrete metric API
+    for metric_method_name in metrics.__all__:
+        # excludes plot_* methods
+        # exclude class (e.g. metrics.ConfusionMatrixDisplay)
+        metric_method = getattr(metrics, metric_method_name)
+        if not inspect.isclass(metric_method) and callable(metric_method) \
+                and not metric_method_name.startswith('plot_'):
+            safe_patch(FLAVOR_NAME, metrics, metric_method_name, patched_metric_api, manage_run=False)
```

Comment:
Have you check this page: https://scikit-learn.org/stable/modules/classes.html#model-selection-interface? I think we need to exclude the following functions:

- `metrics.check_scoring(estimator[, scoring, …])`
- `metrics.get_scorer(scoring) | Get a scorer from string`
- `metrics.make_scorer(score_func, *[, …])`





metrics.check_scoring(estimator[, scoring, …]) | Determine scorer from user options.
-- | --
metrics.get_scorer(scoring) | Get a scorer from string.
metrics.make_scorer(score_func, *[, …]) | Make a scorer from a performance metric or loss function.

Reply:
Good catch :)

Reply:
Shall we use a whitelist for metric API to be patched ? (It would be safer)

Reply:
A TODO: 
we need also patch the scorers in `sklearn.metrics.SCORES`